File: backend/internal/api/workout/workouts.go
package workout

import (
	"encoding/json"
	"net/http"
	"sync"

	"github.com/Scxtto/Hearth/backend/internal/models"
)

// In-memory store for MVP
var (
	workouts []models.Workout
	wMtx     sync.Mutex
)

// RegisterWorkoutRoutes sets up HTTP handlers for workouts
func RegisterWorkoutRoutes(mux *http.ServeMux) {
	mux.HandleFunc("/api/workouts", workoutsHandler)
}

// Handles GET/POST for /api/workouts
func workoutsHandler(w http.ResponseWriter, r *http.Request) {
	switch r.Method {
	case http.MethodGet:
		getWorkouts(w, r)
	case http.MethodPost:
		createWorkout(w, r)
	default:
		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
	}
}

// GET /api/workouts
func getWorkouts(w http.ResponseWriter, r *http.Request) {
	wMtx.Lock()
	defer wMtx.Unlock()
	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(workouts)
}

// POST /api/workouts
func createWorkout(w http.ResponseWriter, r *http.Request) {
	wMtx.Lock()
	defer wMtx.Unlock()
	var session models.Workout
	if err := json.NewDecoder(r.Body).Decode(&session); err != nil {
		http.Error(w, "Invalid JSON: "+err.Error(), http.StatusBadRequest)
		return
	}
	session.ID = len(workouts) + 1 // Basic ID for now
	workouts = append(workouts, session)
	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(http.StatusCreated)
	json.NewEncoder(w).Encode(session)
}

File: frontend/src/services/workouts.js
const API_BASE_URL = "http://localhost:8080/api"; // will be http://localhost:8080/api

export async function saveWorkoutSession(session) {
    console.log("Saving workout session:", session);
    console.log("API Base URL:", API_BASE_URL);
  const res = await fetch(`${API_BASE_URL}/workouts`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(session),
  });
  if (!res.ok) throw new Error("Failed to save workout");
  return await res.json();
}

export async function fetchWorkouts() {
  const res = await fetch(`${API_BASE_URL}/workouts`);
  if (!res.ok) throw new Error("Failed to fetch workouts");
  return await res.json();
}